components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java

```diff
@@ -222,6 +233,54 @@ private void deployClassMediators(List<Artifact.Dependency> artifacts,
     }
 
     /**
+     * Deploy synapse libraries contains in the CApp
+     *
+     * @param artifacts  List of Artifacts contains in the capp
+     * @param axisConfig AxisConfiguration of the current tenant
+     * @throws DeploymentException if something goes wrong while deployment
+     */
+    private void deploySynapseLibrary(List<Artifact.Dependency> artifacts,
+                                      AxisConfiguration axisConfig) throws DeploymentException {
+        for (Artifact.Dependency dependency : artifacts) {
+
+            Artifact artifact = dependency.getArtifact();
+            if (!validateArtifact(artifact)) {
+                continue;
+            }
+
+            if (SynapseAppDeployerConstants.SYNAPSE_LIBRARY_TYPE.equals(artifact.getType())) {
+
+                Deployer deployer = getSynapseLibraryDeployer(axisConfig);
+
+                if (deployer != null) {
+                    artifact.setRuntimeObjectName(artifact.getName());
+                    String fileName = artifact.getFiles().get(0).getName();
+                    String artifactPath = artifact.getExtractedPath() + File.separator + fileName;
+
+                    try {
+                        deployer.deploy(new DeploymentFileData(new File(artifactPath), deployer));
+                        DeploymentFileData dfd = new DeploymentFileData(new File(artifactPath), deployer);
+                        artifact.setDeploymentStatus(AppDeployerConstants.DEPLOYMENT_STATUS_DEPLOYED);
+                        MediationLibraryAdminService mediationLibraryAdminService = new MediationLibraryAdminService();
+                        try {
+                            String artifactName = mediationLibraryAdminService.getArtifactName(artifactPath);
```

Comment:
The method getArtifactName() should use getSynapseConfiguration() (written in 
AbstractServiceBusAdmin). So that I wrote the method in MediationLibraryAdminService.